docs/source/graph-construction/gs-processing/gspartition/sagemaker.rst

```diff
@@ -0,0 +1,134 @@
+
+The following command will authenticate the user account to access to user's ECR repository via AWS CLI.
+
+.. code-block:: bash
+
+    aws ecr get-login-password --region <REGION> | docker login --username AWS --password-stdin <AWS_ACCOUNT_ID>.dkr.ecr.<REGION>.amazonaws.com
+
+Please replace the `<REGION>` and `<AWS_ACCOUNT_ID>` with your own account information and be consistent with the values used in the **Step 1**.
+
+In addition, users need to create an ECR repository at the specified `<REGION>` with the name as `<IMAGE_NAME>` **WITHOUT** the ECR URI string, e.g., ``graphstorm``.
+
+And then use the following command to push the built GraphStorm Docker image to users' own ECR repository.
+
+.. code-block:: bash
+
+    docker push <IMAGE_NAME>:<IMAGE_TAG>
+
+Please replace the `<IMAGE_NAME>` and `<IMAGE_TAG>` with the actual Docker image name and tag, e.g., ``888888888888.dkr.ecr.us-east-1.amazonaws.com/graphstorm:sm-gpu``.
+
+Launch the partition job on Amazon SageMaker
+---------------------------------------------
+For this example, we'll use a SageMaker cluster with 2 ``ml.t3.xlarge`` instances.
+We assume the data is already ready on an AWS S3 bucket.
+For large graph, users can choose larger instances or more instances.
+
+Install dependencies
+.....................
+To run GraphStorm with the Amazon SageMaker service, users should install SageMaker library and copy GraphStorm's SageMaker tools.
+
+1. Use the below command to install SageMaker.
+
+.. code-block:: bash
+
+    pip install sagemaker
+
+2. Copy GraphStorm SageMaker tools. Users can clone the GraphStorm repository using the following command or copy the `sagemaker folder <https://github.com/awslabs/graphstorm/tree/main/sagemaker>`_ to the instance.
+
+.. code-block:: bash
+
+    git clone https://github.com/awslabs/graphstorm.git
+
+Launch graph partitioning task
+...............................
+Users can use the following command to launch partition jobs.
+
+.. code:: bash
+
+   python launch/launch_partition.py \
+       --graph-data-s3 ${DATASET_S3_PATH} \
+       --num-parts ${NUM_PARTITIONS} \
+       --instance-count ${NUM_INSTANCES} \
+       --output-data-s3 ${OUTPUT_PATH} \
+       --instance-type ${INSTANCE_TYPE} \
+       --image-url ${IMAGE_URI} \
+       --region ${REGION} \
+       --role ${ROLE}  \
+       --entry-point "run/partition_entry.py" \
+       --metadata-filename ${METADATA_FILE} \
+       --log-level INFO \
+       --partition-algorithm ${ALGORITHM}
+
+.. warning::
+    The ``NUM_INSTANCES`` should be equal to ``NUM_PARTITIONS`` here.
+
+Running the above will take the dataset after GSProcessing
+from ``${DATASET_S3_PATH}`` as input and create a DistDGL graph with
+``${NUM_PARTITIONS}`` under the output path, ``${OUTPUT_PATH}``.
+Currently we only support ``random`` as the partitioning algorithm for sagemaker.
```

Comment:
When will we support others?

Reply:
For Parmetis, we plan to support it in the next release.